R/aggregate.R

```diff
@@ -180,43 +143,380 @@ agg_climate_data <- function(year, data_raster, climate_var, daily_agg, trans =
+  }
+
+  if(is.null(center_on) & is.null(start_on) & is.null(end_on)){
+    message(crayon::yellow("No bin placement argument specified. Drawing bins from min value."))
+    start_on <- min
+  }
+
+  # Put start_on and end_on in terms of center_on
+  if(!is.null(start_on)){
+    center_on <- start_on + (binwidth / 2)
+  }
+
+  if(!is.null(end_on)){
+    center_on <- end_on - (binwidth / 2)
+  }
+
+
+
+  # Check to see that center_on falls within the range of data, move it if not
+  if(max - min < binwidth){
+    stop(crayon::red("The binwidth is larger than the range of data. Please specify a different binwidth and rerun"))
+  }
+
+  if(center_on > max){
+    message(crayon::yellow("Bin center is greater than maximum value in data. Adjusting it by a multiple of the binwidth so that the bin center falls within the range of data"))
+
+    while(center_on > max){
+      center_on <- center_on - binwidth
+    }
+  }
+  if(center_on < min){
+    message(crayon::yellow("Bin center is less than minimum value in data. Adjusting it by a multiple of the binwidth so that the bin center falls within the range of data"))
+
+    while(center_on < min){
+      center_on <- center_on + binwidth
+    }
+  }
+
+
+  # Create table of bins
+  center <- c(center_on)
+  while(max(center) + (binwidth / 2) < max){
+    center <- c(center, (max(center) + binwidth))
+  }
+  while(min(center) - (binwidth / 2) > min){
+    center <- c(center, (min(center)) - binwidth)
+  }
+
+  center <- sort(center)
+
+
+
+  bins_table <- data.table::data.table(center)
+  bins_table[, ':=' (start = center - (binwidth / 2), end = center + (binwidth / 2))]
+
+  # Readjust max if bin boundaries don't line up properly
+  if(max(bins_table) > max){
+    max <- max(bins_table$end)
+
+    message(crayon::yellow("Max value increased to fit all bins"))
+  }
+  if(min(bins_table) < min){
+    min <- min(bins_table$start)
+    message(crayon::yellow("Min value decreased to fit all bins"))
+  }
+
+  # Readjust number of bins in case the bin boundary's failure to line up cause the creation of an extra bin
+  num_bins <- nrow(bins_table)
+
+  # The bins_table created lists center, start, and end of all bins in order
+
+
+  # Create names for new columns
+  list_names <- sapply(0:(num_bins + 1), FUN=function(x){
+    if(x == 0){
+      paste("-inf", "to", min, sep = "_")
+    }
+    else if(x == num_bins + 1){
+      paste(max, "to", "inf", sep = "_")
+    }
+    else{
+      paste(bins_table[x, start], "to", bins_table[x, end], sep = "_")
+      }
+    })
+
+
+  # Function check_bins to determine which bins data points fall into
+  check_bins <- function(x){
+    clim_daily_table <- raster::values(clim_daily)
+
+    if(x == 0){
+      clim_daily_table <- ifelse(min > clim_daily_table, 1, 0)
+    }
+    else if(x == num_bins + 1){
+      clim_daily_table <- ifelse(max < clim_daily_table, 1, 0)
+    }
+    else{
+      clim_daily_table <- ifelse(bins_table[x, start] <= clim_daily_table &
```

Comment:
you need one side of this to always be a strict inequality and one to be a weak inequality. i don't care which, but it has to be one and it should be clearly documented (e.g., 5 falls into the 0-5 bin centered at 2.5 but not the 5-10 bin centered at 7.5)

Reply:
Good catch I'll make sure to change that